### DISCUSSION TITLE
SMTP ISSUE with sendgrid smtp setup

### DISCUSSION THREAD

**Discussion by Anim12** [chatwoot/chatwoot#5437] [Chatwoot APIs]
I am facing issue 
->while smtp setup with sendgrid
->Not able to connect twitter account getting error occured
->And error message service unavailable after connecting facebook
->Telegram Channel not receiving bot message.

2022-09-15T04:48:21.749Z pid=1 tid=2evt class=ActionMailer::MailDeliveryJob jid=aa539a1bceb50486a4d11ec3 elapsed=0.852 INFO: fail
2022-09-15T04:48:21.749Z pid=1 tid=2evt WARN: {"context":"Job raised exception","job":{"retry":true,"queue":"mailers","class":"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper","wrapped":"ActionMailer::MailDeliveryJob","args":[{"job_class":"ActionMailer::MailDeliveryJob","job_id":"ba8b991d-abbd-484f-96ce-412140625ffb","provider_job_id":null,"queue_name":"mailers","priority":null,"arguments":["Devise::Mailer","reset_password_instructions","deliver_now",{"params":{"account":null,"_aj_symbol_keys":["account"]},"args":[{"_aj_globalid":"gid://chatwoot/User/5"},"sSqBvBXKx4s2kxgLuts1",{"client_config":"default","_aj_symbol_keys":["client_config"]}],"_aj_ruby2_keywords":["params","args"]}],"executions":0,"exception_executions":{},"locale":"en","timezone":"UTC","enqueued_at":"2022-09-15T04:47:38Z"}],"jid":"aa539a1bceb50486a4d11ec3","created_at":1663217258.8655543,"enqueued_at":1663217300.897018,"error_message":"550 The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity requirements\n","error_class":"Net::SMTPFatalError","failed_at":1663217260.1350918,"retry_count":1,"retried_at":1663217278.9257562},"jobstr":"{\"retry\":true,\"queue\":\"mailers\",\"class\":\"ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper\",\"wrapped\":\"ActionMailer::MailDeliveryJob\",\"args\":[{\"job_class\":\"ActionMailer::MailDeliveryJob\",\"job_id\":\"ba8b991d-abbd-484f-96ce-412140625ffb\",\"provider_job_id\":null,\"queue_name\":\"mailers\",\"priority\":null,\"arguments\":[\"Devise::Mailer\",\"reset_password_instructions\",\"deliver_now\",{\"params\":{\"account\":null,\"_aj_symbol_keys\":[\"account\"]},\"args\":[{\"_aj_globalid\":\"gid://chatwoot/User/5\"},\"sSqBvBXKx4s2kxgLuts1\",{\"client_config\":\"default\",\"_aj_symbol_keys\":[\"client_config\"]}],\"_aj_ruby2_keywords\":[\"params\",\"args\"]}],\"executions\":0,\"exception_executions\":{},\"locale\":\"en\",\"timezone\":\"UTC\",\"enqueued_at\":\"2022-09-15T04:47:38Z\"}],\"jid\":\"aa539a1bceb50486a4d11ec3\",\"created_at\":1663217258.8655543,\"enqueued_at\":1663217300.897018,\"error_message\":\"550 The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity requirements\\n\",\"error_class\":\"Net::SMTPFatalError\",\"failed_at\":1663217260.1350918,\"retry_count\":1,\"retried_at\":1663217278.9257562}"}
2022-09-15T04:48:21.749Z pid=1 tid=2evt WARN: Net::SMTPFatalError: 550 The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity requirements

2022-09-15T04:48:21.749Z pid=1 tid=2evt WARN: /usr/local/lib/ruby/3.0.0/net/smtp.rb:1003:in `check_response'
/usr/local/lib/ruby/3.0.0/net/smtp.rb:948:in `data'
/usr/local/lib/ruby/3.0.0/net/smtp.rb:699:in `block in send_message'
/usr/local/lib/ruby/3.0.0/net/smtp.rb:891:in `rcptto_list'
/usr/local/lib/ruby/3.0.0/net/smtp.rb:699:in `send_message'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/network/delivery_methods/smtp_connection.rb:54:in `deliver!'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/network/delivery_methods/smtp.rb:101:in `block in deliver!'
/usr/local/lib/ruby/3.0.0/net/smtp.rb:558:in `start'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/network/delivery_methods/smtp.rb:109:in `start_smtp_session'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/network/delivery_methods/smtp.rb:100:in `deliver!'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/message.rb:2159:in `do_delivery'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/message.rb:260:in `block in deliver'
/gems/ruby/3.0.0/gems/actionmailer-6.1.6.1/lib/action_mailer/base.rb:559:in `block in deliver_mail'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/notifications.rb:203:in `block in instrument'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/notifications.rb:203:in `instrument'
/gems/ruby/3.0.0/gems/actionmailer-6.1.6.1/lib/action_mailer/base.rb:557:in `deliver_mail'
/gems/ruby/3.0.0/gems/mail-2.7.1/lib/mail/message.rb:260:in `deliver'
/gems/ruby/3.0.0/gems/actionmailer-6.1.6.1/lib/action_mailer/message_delivery.rb:119:in `block in deliver_now'
/gems/ruby/3.0.0/gems/actionmailer-6.1.6.1/lib/action_mailer/rescuable.rb:17:in `handle_exceptions'
/gems/ruby/3.0.0/gems/actionmailer-6.1.6.1/lib/action_mailer/message_delivery.rb:118:in `deliver_now'
/gems/ruby/3.0.0/gems/actionmailer-6.1.6.1/lib/action_mailer/mail_delivery_job.rb:23:in `perform'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/execution.rb:48:in `block in perform_now'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:117:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/i18n-1.11.0/lib/i18n.rb:322:in `with_locale'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/translation.rb:9:in `block (2 levels) in <module:Translation>'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `instance_exec'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/core_ext/time/zones.rb:66:in `use_zone'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/timezones.rb:9:in `block (2 levels) in <module:Timezones>'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `instance_exec'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/instrumentation.rb:21:in `block in instrument'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/notifications.rb:203:in `block in instrument'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/notifications.rb:203:in `instrument'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/instrumentation.rb:31:in `instrument'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/instrumentation.rb:14:in `block (2 levels) in <module:Instrumentation>'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `instance_exec'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/logging.rb:22:in `block in tag_logger'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/tagged_logging.rb:99:in `block in tagged'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/tagged_logging.rb:37:in `tagged'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/tagged_logging.rb:99:in `tagged'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/logging.rb:22:in `tag_logger'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/logging.rb:15:in `block (2 levels) in <module:Logging>'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `instance_exec'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:137:in `run_callbacks'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/execution.rb:47:in `perform_now'
/gems/ruby/3.0.0/gems/sentry-rails-5.3.1/lib/sentry/rails/active_job.rb:6:in `perform_now'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/execution.rb:25:in `block in execute'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:117:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/railtie.rb:47:in `block (4 levels) in <class:Railtie>'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/execution_wrapper.rb:91:in `wrap'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/reloader.rb:72:in `block in wrap'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/execution_wrapper.rb:87:in `wrap'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/reloader.rb:71:in `wrap'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/railtie.rb:46:in `block (3 levels) in <class:Railtie>'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `instance_exec'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:126:in `block in run_callbacks'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/callbacks.rb:137:in `run_callbacks'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/execution.rb:23:in `execute'
/gems/ruby/3.0.0/gems/activejob-6.1.6.1/lib/active_job/queue_adapters/sidekiq_adapter.rb:42:in `perform'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:196:in `execute_job'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:164:in `block (2 levels) in process'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/middleware/chain.rb:138:in `block in invoke'
/gems/ruby/3.0.0/gems/sentry-sidekiq-5.3.1/lib/sentry/sidekiq/sentry_context_middleware.rb:7:in `call'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/middleware/chain.rb:140:in `block in invoke'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/middleware/chain.rb:143:in `invoke'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:163:in `block in process'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:136:in `block (6 levels) in dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/job_retry.rb:114:in `local'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:135:in `block (5 levels) in dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/rails.rb:14:in `block in call'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/execution_wrapper.rb:91:in `wrap'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/reloader.rb:72:in `block in wrap'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/execution_wrapper.rb:91:in `wrap'
/gems/ruby/3.0.0/gems/activesupport-6.1.6.1/lib/active_support/reloader.rb:71:in `wrap'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/rails.rb:13:in `call'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:131:in `block (4 levels) in dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:257:in `stats'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:126:in `block (3 levels) in dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/job_logger.rb:13:in `call'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:125:in `block (2 levels) in dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/job_retry.rb:81:in `global'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:124:in `block in dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/job_logger.rb:39:in `prepare'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:123:in `dispatch'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:162:in `process'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:78:in `process_one'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/processor.rb:68:in `run'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/util.rb:56:in `watchdog'
/gems/ruby/3.0.0/gems/sidekiq-6.4.2/lib/sidekiq/util.rb:65:in `block in safe_thread'

**Comment by sojan-official** [comment]
Please check the error message. 
```
The from address does not match a verified Sender Identity. Mail cannot be sent until this error is resolved. Visit https://sendgrid.com/docs/for-developers/sending-email/sender-identity/ to see the Sender Identity.
```

Verify whether the `MAILER_SENDER_EMAIL` is verified in sendgrid.

**Reply by Anim12** [reply]
Its verified.But getting same error.

**Reply by mrrazahussain** [reply]
did you find soution. @Anim12 ?

**Comment by aureliolk** [comment]
Hi guys I'm facing this problem.

How can I resolve? I have already applied domain validation in I sendgrid.

But the problem continues.

![image](https://github.com/chatwoot/chatwoot/assets/18712107/ba387f58-8f84-4c32-8095-cfac9965302b)

![image](https://github.com/chatwoot/chatwoot/assets/18712107/94110e16-f3ca-45b8-bc49-606b67757108)

help!

**Reply by PtkFerraro** [reply]
Did you do work? how? thanks...
